scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+    int totalResults?;
+    int startIndex?;
+    int itemsPerPage?;
+    string[] schemas?;
+    UserResource[] Resources?;
+};
+
+# Represents the error response
+#
+# + detail - Detailed error description  
+# + status - HTTP status code 
+# + schemas - URIs of the used SCIM schemas  
+# + scimType - SCIM type
+public type ErrorResponse record {
+    string detail;
+    string status;
+    string[] schemas;
+    string scimType?;
+};
+
+# Represents the response of the createUser and getUser operations.
+#
+# + schemas - URIs of the used SCIM schemas
+# + id - Unique identifier by service provider  
+# + externalId - Identifier by the provisioning client   
+# + userName - Unique identifier for the user 
+# + name - Name of the user  
+# + displayName - Name displayed to end-users
+# + nickName - Casual name of the user  
+# + profileUrl - URI of the user's online profile  
+# + emails - Email addresses for the User  
+# + addresses - Physical mailing address for the user  
+# + phoneNumbers - Phone numbers for the user  
+# + photos - URI of image of the user  
+# + locale - Default location of the user  
+# + active - Administrative status of the user 
+# + meta - Metadata of the user 
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema  
+# + urn\:scim\:wso2\:schema - Attributes belong to WSO2 custom extension schema  
+# + roles - List of roles for the user
+public type UserResource record {
+    string[]|string schemas?;
+    string id?;
+    string externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    string[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Photo[] photos?;
+    string locale?;
+    boolean active?;
+    Meta meta;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    Custom urn\:scim\:wso2\:schema?;
+    Role[] roles?;
+};
+
+# Represents the sub-attributes of the roles complex attribute in SCIMUser record.
+#
+# + display - Human readable name, primarily used for display purposes
+# + value - Label representing a collection of entitlements  
+# + \$ref - Reference URI 
+# + resourceType - Resource type of the entitlement
+public type Role record {
+    string display;
+    string value?;
+    string \$ref?;
+    string resourceType?;
+};
+
+# Represents the sub-attributes of the meta complex attribute in SCIMUser record.
+#
+# + created - DateTime that the resource was added 
+# + location - URI of the resource being returned 
+# + lastModified - Most recent DateTime that the details of this resource were updated  
+# + resourceType - Name of the resource type of the resource  
+# + 'version - Version of the resource being returned
+public type Meta record {
+    string created;
+    string location;
+    string lastModified;
+    string resourceType?;
+    string 'version?;
+};
```

Comment:
Why this meta record is only defined in userRecord?. It seems the Group record also refers to this Meta record. Can we move the Meta record to a separate Record file.  IMO, it will be more cleaner. WDYT?
https://www.rfc-editor.org/rfc/rfc7643.html#section-3.1 core attributes are allowed for any scim resource type.

s